charset_normalizer/cd.py

```diff
@@ -32,9 +33,18 @@ def encoding_unicode_range(iana_name: str) -> List[str]:
                 continue
 
             if is_unicode_range_secondary(character_range) is False:
-                seen_ranges.add(character_range)
+                if character_range not in seen_ranges:
+                    seen_ranges[character_range] = 0
+                seen_ranges[character_range] += 1
+                character_count += 1
 
-    return sorted(list(seen_ranges))
+    return sorted(
+        [
+            character_range
+            for character_range in seen_ranges
+            if seen_ranges[character_range] / character_count >= 0.15
```

Comment:
That ratio is based on reasonable observations made with all the python core supported CPs